numpy/core/tests/test_umath.py

```diff
@@ -1114,7 +1144,175 @@ def test_reciprocal_values(self):
 
         with np.errstate(divide='raise'):
             for dt in ['f', 'd', 'g']:
-                assert_raises(FloatingPointError, np.reciprocal, np.array(-0.0, dtype=dt))
+                assert_raises(FloatingPointError, np.reciprocal,
+                              np.array(-0.0, dtype=dt))
+
+    def test_tan(self):
+        with np.errstate(all='ignore'):
+            in_ = [np.nan, -np.nan, 0.0, -0.0, np.inf, -np.inf]
+            out = [np.nan, np.nan, 0.0, -0.0, np.nan, np.nan]
+            for dt in ['f', 'd']:
+                in_arr = np.array(in_, dtype=dt)
+                out_arr = np.array(out, dtype=dt)
+                assert_equal(np.tan(in_arr), out_arr)
+
+        with np.errstate(invalid='raise'):
+            for dt in ['f', 'd']:
+                assert_raises(FloatingPointError, np.tan,
+                              np.array(np.inf, dtype=dt))
+                assert_raises(FloatingPointError, np.tan,
+                              np.array(-np.inf, dtype=dt))
+
+    def test_arcsincos(self):
+        with np.errstate(all='ignore'):
+            in_ = [np.nan, -np.nan, np.inf, -np.inf]
+            out = [np.nan, np.nan, np.nan, np.nan]
+            for dt in ['f', 'd']:
+                in_arr = np.array(in_, dtype=dt)
+                out_arr = np.array(out, dtype=dt)
+                assert_equal(np.arcsin(in_arr), out_arr)
+                assert_equal(np.arccos(in_arr), out_arr)
+
+        with np.errstate(invalid='raise'):
+            for dt in ['f', 'd']:
+                assert_raises(FloatingPointError, np.arcsin,
+                              np.array(np.inf, dtype=dt))
+                assert_raises(FloatingPointError, np.arcsin,
+                              np.array(-np.inf, dtype=dt))
+                assert_raises(FloatingPointError, np.arcsin,
+                              np.array(2.0, dtype=dt))
+                assert_raises(FloatingPointError, np.arcsin,
+                              np.array(-2.0, dtype=dt))
+                assert_raises(FloatingPointError, np.arccos,
+                              np.array(np.inf, dtype=dt))
+                assert_raises(FloatingPointError, np.arccos,
+                              np.array(-np.inf, dtype=dt))
+                assert_raises(FloatingPointError, np.arccos,
+                              np.array(2.0, dtype=dt))
+                assert_raises(FloatingPointError, np.arccos,
+                              np.array(-2.0, dtype=dt))
```

Comment:
Would it maybe make sense to iterate over the callable here since the top/bottom 4 `assert_raises` calls are the same except for the callable?

i.e., something like:

```python
for callable in [np.arcsin, np.arccos]:
    for dt in ['f', 'd']:
       assert_raises(FloatingPointError, callable,
                           args)
   ...
```

Reply:
Makes sense. I will fix that.